tests/model/test_cycle_data.py

```diff
@@ -253,48 +247,144 @@ def test_cycle_performance_data(
-        assert PerformanceSignature.objects.count() == 2
+    assert list(PerformanceDatum.objects.values_list('id', flat=True)) == [1]
+    assert list(PerformanceSignature.objects.values_list('id', flat=True)) == [
+        test_perf_signature.id
+    ]
 
 
-def test_performance_cycler_quit_indicator():
-    ten_minutes_ago = datetime.datetime.now() - datetime.timedelta(minutes=10)
-    max_one_second = datetime.timedelta(seconds=1)
-
-    two_seconds_ago = datetime.datetime.now() - datetime.timedelta(seconds=2)
-    max_five_minutes = datetime.timedelta(minutes=5)
-
-    with pytest.raises(MaxRuntimeExceeded):
-        PerformanceDatumManager._maybe_quit(
-            started_at=ten_minutes_ago, max_overall_runtime=max_one_second
+def test_try_data_removal(
+    try_repository, test_repository, try_push_stored, test_perf_signature, test_perf_signature_2
+):
+    total_removals = 3
+    test_perf_signature.repository = try_repository
+    test_perf_signature.save()
+
+    try_pushes = list(Push.objects.filter(repository=try_repository).order_by('id').all())
+
+    for idx, push in enumerate(try_pushes[:-2]):
+        push_timestamp = datetime.now()
+        if idx < total_removals:
+            push_timestamp -= timedelta(weeks=10)
+
+        PerformanceDatum.objects.create(
+            repository=try_repository,
+            push=push,
+            job=None,
+            signature=test_perf_signature,
+            push_timestamp=push_timestamp,
+            value=1.0,
         )
 
-    try:
-        PerformanceDatumManager._maybe_quit(
-            started_at=two_seconds_ago, max_overall_runtime=max_five_minutes
+    for push in try_pushes[-2:]:
+        push_timestamp = datetime.now() - timedelta(weeks=10)
+
+        # try data removal shouldn't delete these non-try data
+        PerformanceDatum.objects.create(
+            repository=test_repository,
+            push=push,
+            job=None,
+            signature=test_perf_signature_2,
+            push_timestamp=push_timestamp,
+            value=1.0,
         )
-    except MaxRuntimeExceeded:
-        pytest.fail('Performance cycling shouldn\'t have quit')
 
+    total_initial_data = PerformanceDatum.objects.count()
+
+    call_command('cycle_data', 'from:perfherder')
+    assert PerformanceDatum.objects.count() == total_initial_data - total_removals
+    assert not PerformanceDatum.objects.filter(
+        push_timestamp__lt=datetime.now() - timedelta(weeks=6),
+        repository=try_repository,
+    ).exists()
+    assert (
+        PerformanceDatum.objects.exclude(repository=try_repository).count() == 2
+    )  # non-try data remained intact
+
+
+@pytest.mark.parametrize(
+    'repository_name',
+    ['autoland', 'mozilla-beta', 'mozilla-central', 'fenix', 'reference-browser'],
+)
+def test_irrelevant_repos_data_removal(
+    test_repository,
+    relevant_repository,
+    repository_name,
+    push_stored,
+    test_perf_signature,
+    test_perf_signature_2,
+):
```

Comment:
Your new strategy is dependent upon the existence of the repositories in the database.
You need to provide those as fixture before this test run.